controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -180,14 +180,28 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 	}
 
 	// Add KMS details to Noobaa spec, only if
-	// cluster-wide encryption is enabled (ie; sc.Spec.Encryption.Enable is True)
+	// cluster-wide encryption is enabled (ie; sc.Spec.Encryption.ClusterWide is True)
 	// and KMS ConfigMap is available
-	if sc.Spec.Encryption.Enable {
+	if sc.Spec.Encryption.ClusterWide {
 		if kmsConfig, err := getKMSConfigMap(KMSConfigMapName, sc, r.Client); err != nil {
 			return err
 		} else if kmsConfig != nil {
+			// Set default KMS_PROVIDER. Possible values are: vault, ibmkeyprotect.
+			if _, ok := kmsConfig.Data["KMS_PROVIDER"]; !ok {
```

Comment:
Both are not exactly the same, even if I am moving to a common function for a few fields we still do not do a lot of if-else conditions, and it may keep changing in the future.